pwndbg/aglib/regs.py

```diff
@@ -0,0 +1,296 @@
+"""
+Reading register value from the inferior, and provides a
+standardized interface to registers like "sp" and "pc".
+"""
+
+from __future__ import annotations
+
+import ctypes
+import re
+import sys
+from types import ModuleType
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+from typing import cast
+
+import pwndbg
+import pwndbg.aglib.arch
+import pwndbg.aglib.proc
+import pwndbg.aglib.qemu
+import pwndbg.aglib.remote
+import pwndbg.aglib.typeinfo
+import pwndbg.lib.cache
+from pwndbg.dbg import EventType
+from pwndbg.lib.regs import BitFlags
+from pwndbg.lib.regs import RegisterSet
+from pwndbg.lib.regs import reg_sets
+
+
+@pwndbg.lib.cache.cache_until("stop", "prompt")
+def regs_in_frame(frame: pwndbg.dbg_mod.Frame) -> pwndbg.dbg_mod.Registers:
+    return frame.regs()
+
+
+@pwndbg.aglib.proc.OnlyWhenRunning
+def get_register(
+    name: str, frame: pwndbg.dbg_mod.Frame | None = None
+) -> pwndbg.dbg_mod.Value | None:
+    if frame is None:
+        frame = pwndbg.dbg.selected_frame()
+    assert (
+        frame is not None
+    ), "pwndbg.dbg.selected_frame() should never return None when marked with @OnlyWhenRunning"
+
+    regs = regs_in_frame(frame)
+
+    return regs.by_name(name) or regs.by_name(name.upper())
```

Comment:
If register value is 0, what does the `bool(pwndbg.dbg_mod.Value)` return?

If it is false, then we are effectively fetching the same value again unnecessarily.

https://github.com/pwndbg/pwndbg/blob/27ccbbaceb5b559b3e07bddfc9616b7da50bab07/pwndbg/gdblib/regs.py#L35-L42

Reply:
`pwndbg.dbg_mod.Value` implements neither `__bool__` or `__len__`, so, `bool()` of it [will be `True` for all valid instances](https://docs.python.org/3.8/reference/datamodel.html#object.__bool__).